This was after pulling a fresh zip copy. Just tried again on a different computer, no luck.
No errors at first but in the output I get tons of this kinda thing:

> 3>CSC : error CS2001: Source file 'D:\Dev\SDK\TEMP\Avalonia\src\Markup\Avalonia.Markup.Xaml\OmniXAML\Source\OmniXaml\ObjectAssembler\NullLevel.cs' could not be found.
> 3>CSC : error CS2001: Source file 'D:\Dev\SDK\TEMP\Avalonia\src\Markup\Avalonia.Markup.Xaml\OmniXAML\Source\OmniXaml\Parsers\ProtoParser\AttributeParser.cs' could not be found.
> 3>CSC : error CS2001: Source file 'D:\Dev\SDK\TEMP\Avalonia\src\Markup\Avalonia.Markup.Xaml\OmniXAML\Source\OmniXaml\IParserFactory.cs' could not be found.
> 3>CSC : error CS2001: Source file 'D:\Dev\SDK\TEMP\Avalonia\src\Markup\Avalonia.Markup.Xaml\OmniXAML\Source\OmniXaml\Typing\XamlTypeName.cs' could not be found.
> 3>CSC : error CS2001: Source file 'D:\Dev\SDK\TEMP\Avalonia\src\Markup\Avalonia.Markup.Xaml\OmniXAML\Source\OmniXaml\Typing\AttachableMember.cs' could not be found.
> 3>CSC : error CS2001: Source file 'D:\Dev\SDK\TEMP\Avalonia\src\Markup\Avalonia.Markup.Xaml\OmniXAML\Source\OmniXaml\ObjectAssembler\NullLifecycleListener.cs' could not be found.
> 3>CSC : error CS2001: Source file 'D:\Dev\SDK\TEMP\Avalonia\src\Markup\Avalonia.Markup.Xaml\OmniXAML\Source\OmniXaml\IRuntimeTypeSource.cs' could not be found.
> 3>CSC : error CS2001: Source file 'D:\Dev\SDK\TEMP\Avalonia\src\Markup\Avalonia.Markup.Xaml\OmniXAML\Source\OmniXaml\Parsers\ProtoParser\DirectiveAssignment.cs' could not be found.

--- a/src/Windows/Avalonia.Win32/IconImpl.cs
+++ b/src/Windows/Avalonia.Win32/IconImpl.cs
@@ -1,6 +1,7 @@
 ﻿using Avalonia.Platform;
 using System;
 using System.Collections.Generic;
+using System.Drawing;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
@@ -9,11 +10,19 @@ namespace Avalonia.Win32
 {
     class IconImpl : IWindowIconImpl
     {
-        public IconImpl(System.Drawing.Bitmap iconBitmap)
+        private Bitmap bitmap;
+        private Icon icon;
+
+        public IconImpl(Bitmap bitmap)
         {
-            IconBitmap = iconBitmap;
+            this.bitmap = bitmap;
         }
 
-        public System.Drawing.Bitmap IconBitmap { get; }
+        public IconImpl(Icon icon)
+        {
+            this.icon = icon;
+        }
+
+        public IntPtr HIcon => icon?.Handle ?? bitmap.GetHicon();
     }
 }

--- a/src/Windows/Avalonia.Win32/Win32Platform.cs
+++ b/src/Windows/Avalonia.Win32/Win32Platform.cs
@@ -189,14 +189,15 @@ namespace Avalonia.Win32
 
         public IWindowIconImpl LoadIcon(string fileName)
         {
-            var icon = new System.Drawing.Bitmap(fileName);
-            return new IconImpl(icon);
+            using (var stream = File.OpenRead(fileName))
+            {
+                return CreateImpl(stream); 
+            }
         }
 
         public IWindowIconImpl LoadIcon(Stream stream)
         {
-            var icon = new System.Drawing.Bitmap(stream);
-            return new IconImpl(icon);
+            return CreateImpl(stream);
         }
 
         public IWindowIconImpl LoadIcon(IBitmapImpl bitmap)
@@ -205,6 +206,18 @@ namespace Avalonia.Win32
             {
                 bitmap.Save(memoryStream);
                 return new IconImpl(new System.Drawing.Bitmap(memoryStream));
+            }
+        }
+
+        private static IconImpl CreateImpl(Stream stream)
+        {
+            try
+            {
+                return new IconImpl(new System.Drawing.Icon(stream));
+            }
+            catch (ArgumentException)
+            {
+                return new IconImpl(new System.Drawing.Bitmap(stream));
             }
         }
     }

--- a/src/Windows/Avalonia.Win32/WindowImpl.cs
+++ b/src/Windows/Avalonia.Win32/WindowImpl.cs
@@ -684,9 +684,9 @@ namespace Avalonia.Win32
         public void SetIcon(IWindowIconImpl icon)
         {
             var impl = (IconImpl)icon;
-            var nativeIcon = impl.IconBitmap;
+            var hIcon = impl.HIcon;
             UnmanagedMethods.PostMessage(_hwnd, (int)UnmanagedMethods.WindowsMessage.WM_SETICON,
-                new IntPtr((int)UnmanagedMethods.Icons.ICON_BIG), nativeIcon.GetHicon());
+                new IntPtr((int)UnmanagedMethods.Icons.ICON_BIG), hIcon);
         }
 
         private static int ToInt32(IntPtr ptr)

